draft-bradley-dnssd-private-discovery.md

```diff
@@ -69,15 +75,15 @@ document are to be interpreted as described in [@!RFC2119].
 
 Multi-byte values are encoded from the most significant byte to the least significant byte (big endian).
 
-When multiple items are concatenated together, the symbol "||" (without quotes) between each item is used to indicate this. For example, a combined item of A followed by B followed by C would be written as "A || B || C".
+When multiple items are concatenated together, the symbol "\|\|" (without quotes) between each item is used to indicate this. For example, a combined item of A followed by B followed by C would be written as "A \|\| B \|\| C".
 
 # Protocol
 
 There are two techniques used to preserve privacy and provide confidentiality in this document. The first is announcing, probing, and responding with only enough info to allow a peer with your public key to detect that it's you while hiding your identity from peers without your public key. This technique uses a fresh random signed with your private key using a signature algorithm that doesn't reveal your public key. The second technique is to query and answer in a way that only a specific friend can read the data. This uses ephemeral key exchange and symmetric encryption and authentication.
 
 The general flow of the protocol is a device sends multicast probes to discover friend devices on the network. If friend devices are found, it directly communicates with them via unicast queries and answers. Announcements are sent to report availability and when services are added or removed.
 
-Messages use a common header with a flags/type field. This indicates the format of the data after the header. Any data beyond the type-specific message body must be ignored. Future versions of this document may define additional data and this must not cause older message parsers to break. Updated formats that break compatibility with older parsers must use a new message type.
+Messages use a common header with a flags/type field. This indicates the format of the data after the header. Any data beyond the type-specific message body must be ignored. Future versions of this document may define additional data and this must not cause older message parsers to break. Updated formats that break compatibility with older parsers must use a new message type. (That is, there is no explicit version field in the wire image.)
```

Comment:
If we're going to say something about a version field, it might be better to explain. I intentionally excluded a version field I've found version numbers to be a source of compatibility problems. You can't really change a version because it breaks older versions. The flags/type mechanism I use here allows extensions without breaking older code.
```suggestion
Messages use a common header with a flags/type field. This indicates the format of the data after the header. Any data beyond the type-specific message body must be ignored. Future versions of this document may define additional data and this must not cause older message parsers to break. Updated formats that break compatibility with older parsers must use a new message type. (That is, there is no explicit version field in the wire image.)
```
```suggestion
Messages use a common header with a flags/type field. This indicates the format of the data after the header. Any data beyond the type-specific message body must be ignored. Future versions of this document may define additional data and this must not cause older message parsers to break. Updated formats that break compatibility with older parsers must use a new message type. (That is, there is no explicit version field in the wire image.)
```
```suggestion
Messages use a common header with a flags/type field. This indicates the format of the data after the header. Any data beyond the type-specific message body must be ignored. Future versions of this document may define additional data and this must not cause older message parsers to break. Updated formats that break compatibility with older parsers must use a new message type. (That is, there is no explicit version field in the wire image.)
```